File: testdata/jennies/builders/struct_with_defaults/PHPBuilder/src/StructWithDefaults/NestedStructBuilder.php
<?php

namespace Grafana\Foundation\StructWithDefaults;

/**
 * @implements \Grafana\Foundation\Cog\Builder<\Grafana\Foundation\StructWithDefaults\NestedStruct>
 */
class NestedStructBuilder implements \Grafana\Foundation\Cog\Builder
{
    protected \Grafana\Foundation\StructWithDefaults\NestedStruct $internal;

    public function __construct()
    {
    	$this->internal = new \Grafana\Foundation\StructWithDefaults\NestedStruct();
    }

    /**
     * Builds the object.
     * @return \Grafana\Foundation\StructWithDefaults\NestedStruct
     */
    public function build()
    {
        return $this->internal;
    }

    public function stringVal(string $stringVal): static
    {
        $this->internal->stringVal = $stringVal;
    
        return $this;
    }

    public function intVal(int $intVal): static
    {
        $this->internal->intVal = $intVal;
    
        return $this;
    }

}

File: testdata/jennies/builders/struct_with_defaults/JavaBuilder/struct_with_defaults/NestedStructBuilder.java
package struct_with_defaults;


public class NestedStructBuilder implements cog.Builder<NestedStruct> {
    protected final NestedStruct internal;
    
    public NestedStructBuilder() {
        this.internal = new NestedStruct();
    }
    public NestedStructBuilder stringVal(String stringVal) {
        this.internal.stringVal = stringVal;
        return this;
    }
    
    public NestedStructBuilder intVal(Long intVal) {
        this.internal.intVal = intVal;
        return this;
    }
    public NestedStruct build() {
        return this.internal;
    }
}
